### PR TITLE
Add support for calling Parcels.wrap() with a parcelable

### LINKED ISSUE #158: Support for Parcelables in Parcels.wrap
I currently have a bunch of generic methods just calling `Parcels.wrap` when they want to make something parcelable.

However, if the object that I pass in is already a parcelable on its own Parceler fails to find an implementation for that object.
I understand it might be out of scope for the project but I would love it if there was a ParcelableParcelable or something that would make `Parcels.wrap` work with parcelable objects.

### PR DESCRIPTION
**Pull request by ansman**
This fixes #158

### CODE REVIEW

File: examples/test/src/test/java/org/parceler/ParcelTest.java [hunk 1]
```diff
@@ -130,6 +132,13 @@ public void testIntArrayFactoryOrder() {
         assertArrayEquals(new int[]{1, 2, 3, 4, 5}, unwrapped.getArr());
     }
 
+    @Test
+    public void testParcelableClassWithoutAnnotation() {
```
Comment (ansman) [inline examples/test/src/test/java/org/parceler/ParcelTest.java]: Is this the right place to have this test?

**Comment by ansman** [comment]
I have a hard time running the tests for this project, I just keep getting a bunch of `java.lang.NoClassDefFoundError` for android and roboelectric stuff.

For example this:
```
java.lang.NoClassDefFoundError: Could not initialize class org.robolectric.shadows.ShadowMediaMetadataRetriever
```

File: examples/test/src/test/java/org/parceler/ParcelTest.java (hunk 1 above)
Comment (johncarl81) [inline examples/test/src/test/java/org/parceler/ParcelTest.java]: Yeah, that's a fine test for this.  `Intent` seems like a strange use case to test... why not test a specific inner class Parcelable?

File: parceler-api/src/main/java/org/parceler/NonParcelRepository.java [hunk 2]
```diff
@@ -265,6 +265,14 @@ public Parcelable buildParcelable(String input) {
         }
     }
 
+    static class ParcelableParcelableFactory implements Parcels.ParcelableFactory<Parcelable>{
+
+        @Override
+        public Parcelable buildParcelable(Parcelable input) {
+            return new ParcelableParcelable(input);
```
Comment (johncarl81) [inline parceler-api/src/main/java/org/parceler/NonParcelRepository.java]: This feels a little funny, but I guess it takes care of unwrapping the wrapped Parcelable.

File: parceler-api/src/main/java/org/parceler/Parcels.java [hunk 3]
```diff
@@ -194,6 +194,11 @@ public ParcelableFactory get(Class clazz){
             ParcelableFactory result = generatedMap.get(clazz);
             if (result == null) {
                 ParcelableFactory value = findClass(clazz);
+
+                if (Parcelable.class.isAssignableFrom(clazz)) {
```
Comment (johncarl81) [inline parceler-api/src/main/java/org/parceler/Parcels.java]: I'd agree, this is the best place for this conditional.

**Comment by johncarl81** [comment]
Looks like the tests run just fine on my machine and on Travis.

**Comment by johncarl81** [comment]
Before I merge this, I think I need a better use case.  Could you put an example together to demonstrate why we need this?

File: parceler-api/src/main/java/org/parceler/NonParcelRepository.java (hunk 2 above)
Comment (ansman) [inline parceler-api/src/main/java/org/parceler/NonParcelRepository.java]: It's mostly to ensure that if people have done this:
```
ParcelWrapper wrapper = (ParcelWrapper) Parcels.wrap(someObject);
```

It won't crash

File: examples/test/src/test/java/org/parceler/ParcelTest.java (hunk 1 above)
Comment (ansman) [inline examples/test/src/test/java/org/parceler/ParcelTest.java]: No reason, it was just a native Parcelable. But I'd be willing to change if you want to.

**Comment by ansman** [comment]
The reason is that we have mixed @Parcel and Parcelable. It hard and time consuming to find all places where we incorrectly call `Parcels.wrap` with a parcelable. This would just ease the transition.

File: examples/test/src/test/java/org/parceler/ParcelTest.java (hunk 1 above)
Comment (johncarl81) [inline examples/test/src/test/java/org/parceler/ParcelTest.java]: Yeah, if you could add a custom Parcelable, I'd appreciate it.

**Pull request merged**